### DISCUSSION TITLE
Circumvent compose not being called by reactive variable change

### DISCUSSION THREAD

**Discussion by alexisthual** [Textualize/textual#2434] [Q&A] [answered]
Good evening!

First of all, thanks for the great library :blush: 

I am surprised that, in a custom `Widget`, a change in a `reactive` variable doesn't call `compose()` again.
Is there a way that, using `compose()`, I can refresh a widget (and it's children widgets) whose variable just got changed?
If not, can I achieve this without `compose()`?

My use case is the following: I am building an application with multiple nested widgets.
I want a change in one of the main variables of the application to refresh the whole app.
I chose to use `compose()` and not `render()` because I am mostly working with widgets and could not find a way to instantiate lists of widgets in `render()`.

Thanks!

**Comment by davep** [comment, marked as answer]
The way to think of it is `compose` builds the structure of your widget; you don't want to tear down and rebuild a structure every time. If you're using reactives the best thing to do with a compound widget that's build up with `compose` is to use the watch system and update the parts of the widget that require updating.


The documentation talking about [compound widgets](https://textual.textualize.io/guide/widgets/#compound-widgets) might be helpful here.
